### PR TITLE
isascii() function

### PR DESCRIPTION
**Pull request by SadiHassan**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

`isascii()` function.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp [hunk 1]
```diff
@@ -313,6 +313,16 @@ bool IndexOf::convertImpl(String & out, IParser::Pos & pos)
...
 }
 
+bool IsAscii::convertImpl(String & out, IParser::Pos & pos)
+{
+    const String fn_name = getKQLFunctionName(pos);
+    if (fn_name.empty())
+        return false;
+    ++pos;
+    String arg = getConvertedArgument(fn_name, pos);
+    out = std::format("not arrayExists(x -> x < 0 or x > 127, arrayMap(x -> ascii(x), splitByRegexp('', assumeNotNull({}))))", arg);
```
Comment (kashwy) [inline src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp:323]: is `splitByString  `less expensive  than `splitByRegexp' ?

Comment (SadiHassan) [reply src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp:323]: Splitbystring didn't work. It requires a character to split by.

Comment (SadiHassan) [reply src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp]: Sorry, I mixed up `splitByChar` with `splitByString`. Updated code with `splitByString` and it produces same result.

File: src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp [hunk 2]
```diff
@@ -313,6 +313,18 @@ bool IndexOf::convertImpl(String & out, IParser::Pos & pos)
...
 }
 
+
+bool IsAscii::convertImpl(String & out, IParser::Pos & pos)
+{
+    const String fn_name = getKQLFunctionName(pos);
+    if (fn_name.empty())
+        return false;
+    ++pos;
+    String arg = getConvertedArgument(fn_name, pos);
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp:323]:
```suggestion
    const auto arg = getArgument(fn_name, pos);
```

**Review by ltrk2 — CHANGES_REQUESTED** [review]

**Review by kashwy — APPROVED** [review]

**Review by ltrk2 — APPROVED** [review]

**Pull request merged**